sdk/purview/azure-analytics-purview-account/src/main/java/com/azure/analytics/purview/account/PurviewAccountClientBuilder.java

```diff
@@ -0,0 +1,298 @@
+     *
+     * @param tokenCredential the tokenCredential value.
+     * @return the PurviewAccountClientBuilder.
+     */
+    public PurviewAccountClientBuilder credential(TokenCredential tokenCredential) {
+        this.tokenCredential = tokenCredential;
+        return this;
+    }
+
+    /*
+     * The logging configuration for HTTP requests and responses.
+     */
+    private HttpLogOptions httpLogOptions;
+
+    /**
+     * Sets The logging configuration for HTTP requests and responses.
+     *
+     * @param httpLogOptions the httpLogOptions value.
+     * @return the PurviewAccountClientBuilder.
+     */
+    public PurviewAccountClientBuilder httpLogOptions(HttpLogOptions httpLogOptions) {
+        this.httpLogOptions = httpLogOptions;
+        return this;
+    }
+
+    /*
+     * The retry policy that will attempt to retry failed requests, if
+     * applicable.
+     */
+    private RetryPolicy retryPolicy;
+
+    /**
+     * Sets The retry policy that will attempt to retry failed requests, if applicable.
+     *
+     * @param retryPolicy the retryPolicy value.
+     * @return the PurviewAccountClientBuilder.
+     */
+    public PurviewAccountClientBuilder retryPolicy(RetryPolicy retryPolicy) {
+        this.retryPolicy = retryPolicy;
+        return this;
+    }
+
+    /*
+     * The list of Http pipeline policies to add.
+     */
+    private final List<HttpPipelinePolicy> pipelinePolicies;
+
+    /**
+     * Adds a custom Http pipeline policy.
+     *
+     * @param customPolicy The custom Http pipeline policy to add.
+     * @return the PurviewAccountClientBuilder.
+     */
+    public PurviewAccountClientBuilder addPolicy(HttpPipelinePolicy customPolicy) {
+        pipelinePolicies.add(customPolicy);
+        return this;
+    }
+
+    /**
+     * Builds an instance of PurviewAccountClientImpl with the provided parameters.
+     *
+     * @return an instance of PurviewAccountClientImpl.
+     */
+    private PurviewAccountClientImpl buildInnerClient() {
+        if (serviceVersion == null) {
+            this.serviceVersion = PurviewAccountServiceVersion.getLatest();
+        }
+        if (pipeline == null) {
+            this.pipeline = createHttpPipeline();
+        }
+        PurviewAccountClientImpl client =
+                new PurviewAccountClientImpl(
+                        pipeline, JacksonAdapter.createDefaultSerializerAdapter(), endpoint, serviceVersion);
+        return client;
+    }
+
+    private HttpPipeline createHttpPipeline() {
+        Configuration buildConfiguration =
+                (configuration == null) ? Configuration.getGlobalConfiguration() : configuration;
+        if (httpLogOptions == null) {
+            httpLogOptions = new HttpLogOptions();
+        }
+        List<HttpPipelinePolicy> policies = new ArrayList<>();
+        String clientName = properties.getOrDefault(SDK_NAME, "UnknownName");
+        String clientVersion = properties.getOrDefault(SDK_VERSION, "UnknownVersion");
```

Comment:
Not a blocker, but anywhere initialize this `properties` with values?

Reply:
Nowhere. It also happens in existing SDKs. It was last modified in 2020/9 in autorest.java.

Reply:
https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/util/BuilderHelper.java#L48-L49

Reply:
You need to include artifact-id in your autorest config to generate a `*.properties` file which will then initialize the properties map.